modules/admin/api/handler/cms_api_rels_admin_handler.e

```diff
@@ -0,0 +1,357 @@
+note
+	description: "Summary description for {CMS_API_RELS_ADMIN_HANDLER}."
+	date: "$Date$"
+	revision: "$Revision$"
+
+class
+	CMS_API_RELS_ADMIN_HANDLER
+
+inherit
+
+	CMS_HANDLER
+
+	WSF_URI_HANDLER
+		rename
+			execute as uri_execute,
+			new_mapping as new_uri_mapping
+		end
+
+	WSF_URI_TEMPLATE_HANDLER
+		rename
+			execute as uri_template_execute,
+			new_mapping as new_uri_template_mapping
+		select
+			new_uri_template_mapping
+		end
+
+	WSF_RESOURCE_HANDLER_HELPER
+		redefine
+			do_get
+		end
+
+	REFACTORING_HELPER
+
+create
+	make
+
+feature -- execute
+
+	execute (req: WSF_REQUEST; res: WSF_RESPONSE)
+			-- Execute request handler
+		do
+			execute_methods (req, res)
+		end
+
+	uri_execute (req: WSF_REQUEST; res: WSF_RESPONSE)
+			-- Execute request handler
+		do
+			execute (req, res)
+		end
+
+	uri_template_execute (req: WSF_REQUEST; res: WSF_RESPONSE)
+			-- Execute request handler
+		do
+			execute (req, res)
+		end
+
+feature -- HTTP Methods
+
+	do_get (req: WSF_REQUEST; res: WSF_RESPONSE)
+		do
+				-- TODO analize if it's ok to define Rels definitions
+				-- as external HTML files.
+			if attached {WSF_STRING} req.path_parameter ("rel") as l_rel then
+				if l_rel.value.same_string ("api-admin-roles") then
+					compute_response_get (req, res, rel_api_admin_roles, "text/html")
+				elseif l_rel.value.same_string ("api-role-permissions") then
+					compute_response_get (req, res, rel_api_role_permissions, "text/html")
+				elseif l_rel.value.same_string ("api-role-update") then
+					compute_response_get (req, res, rel_api_role_update, "text/html")
+				elseif l_rel.value.same_string ("api-role-delete") then
+					compute_response_get (req, res, rel_api_role_delete, "text/html")
+				end
+			end
+		end
+
+feature -- Rels
+
+feature -- Rels Definitions
+
+	rel_api_admin_roles: STRING = "[
+			<!DOCTYPE html>
+			<html>
+			<head>
+				<link href="/rels.css" media="all" rel="stylesheet" type="text/css">
```

Comment:
you are right, by the way, I don't want to have rels templates inside Eiffel code.
What's your opinion? We can extend hal library to describe rels or we can extract them as templates